feat: 재생 화면 페이지 구현

faber-h opened the pull request:
### ✨ 이슈 번호

- #17 

### 📌 설명

- 재생 화면 UI를 구현하여 작성한 Pull Request입니다.

### 📃 작업 사항

  - [x] 재생 콘텐츠 정보
    - [x] 채널 이미지
    - [x] 채널 제목
    - [x] 채널명
  - [x] 컨트롤러 구성
    - [x] 사용자 클릭 진입 시 광고 감지 토글
    - [x] 자동으로 이동했을 시 기존 채널 이동 토글
  - [x] toggleButton 컴포넌트 size 조절 기능 추가

### 💡 참고 사항

https://github.com/user-attachments/assets/c8e5d93b-4a56-435b-ad08-aeb03e061d21

### 💭 리뷰 사항 반영

- [x] 컴포넌트명에 맞춰 변수명 변경
- [x] ToggleButton 공통 컴포넌트로 사용하여 ui 폴더로 이동
- [x] constant 폴더 확장성 고려하여 constants 폴더로 수정
- [x] isPlaying 사용자 인터랙션으로 재생 시작해야 하므로 초기 값 false로 수정

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

faber-h committed:
feat: 채널 플레이어 UI에 토글 버튼 및 재생/일시정지 기능 추가
- 채널 변경 여부에 따른 버튼 변경
- 버튼 컴포넌트 클릭 이벤트 추가

faber-h committed:
feat: toggleButton 컴포넌트 size 조절 기능 추가

d0-neii commented on src/components/ToggleButton.jsx:
```diff
@@ -1,24 +1,39 @@
 
-const ToggleButton = () => {
-  const [isActive, setIsActive] = useState(false);
+const toggleSize = {
```
`toggleSize`안에 구조가 거의 똑같아서 중복 줄일 수 있는 함수로 바꾸면 더 좋을 것 같습니다!

faber-h replied:
현재는 두 개뿐이라 가독성 우선으로 유지하고, 추후 확장 시 함수로 리팩토링할게요~
좋은 의견 감사합니다!

faber-h committed:
refactor: 불필요한 fragment 구조 제거

d0-neii commented on src/pages/ChannelPlayer.jsx:
```diff
@@ -0,0 +1,52 @@
+  // 데이터 연결 전 임시 값 할당
+  const logoUrl =
+    "https://www.urbanbrush.net/web/wp-content/uploads/edd/2019/08/urbanbrush-20190805082332272597.png";
+  const mainTitle = "RDO 라디오방송";
```
실제 채널 명을 보여주는 거면 prop으로 받고 변수 명도 ChannelPlayer 라는 컴포넌트 명과 어울리게 channelTitle로 바꾸는 게 자연스러울 것 같습니다!

faber-h replied:
처음엔 miniPlayer컴포넌트와 네이밍을 맞추려고 컴포넌트명을 mainPlayer로 작성했다가 채널 특성을 살리자는 생각에 ChannelPlayer로 변경하면서 변수명을 바꾸는 걸 놓쳤네요~
감사합니다!
그리고 채널명 prop는 상태 관리를 위해 zustand 도입을 고려하고 있기 때문에 임시 값은 지역변수로 사용했습니다~

faber-h committed:
refactor: 컴포넌트명에 맞춰 변수명 변경

devjaeyoon commented on src/constant/settingOptions.js:
현재는 `settingOptions.js` 밖에 없지만
추후 다른 상수 파일이 추가될 수 있기 때문에 폴더명은 `constant`보다 `constants`가 더 좋을 것 같습니다.

faber-h replied:
네 확장성을 고려서 `contants`가 더 좋을 것 같네요.
좋은 의견 감사합니다~

devjaeyoon commented on src/pages/ChannelPlayer.jsx:
```diff
@@ -0,0 +1,51 @@
+
+import MainPauseIcon from "@/assets/svgs/icon-main-pause.svg?react";
+import MainPlayIcon from "@/assets/svgs/icon-main-play.svg?react";
+import ToggleButton from "@/components/ToggleButton";
```
`ToggleButton`도 `Button`과 유사한 ui관련 컴포넌트니 `ui` 디렉토리에 들어가는게 좋을 것 같아요.

faber-h replied:
네 공통 컴포넌트로 사용되니까 `ui` 디렉토리로 옮기는 게 좋을 것 같네요.
좋은 의견 감사합니다~

devjaeyoon commented on src/pages/ChannelPlayer.jsx:
```diff
@@ -0,0 +1,51 @@
+  const logoUrl =
+    "https://www.urbanbrush.net/web/wp-content/uploads/edd/2019/08/urbanbrush-20190805082332272597.png";
+  const channelTitle = "RDO 라디오방송";
+  const [isPlaying, setIsPlaying] = useState(true);
```
광고가 재생될 때 다른 채널로 자동 재생하려면 사용자의 직접적인 인터랙션이 최소 1회 필요한데 최소 한 번을 첫 재생 클릭을 한 번 받는게 좋을 것 같으니 `initial value`는 `false`가 더 좋을 것 같습니다.

faber-h replied:
인터랙션 고려하여 초기 값을 `false`로 하는게 더 좋을 것 같네요.
좋은 의견 감사합니다~

devjaeyoon reviewed (changes requested):
리뷰 사항 확인 부탁드립니다.

faber-h committed:
rename: 폴더명 수정, 폴더 이동
- ToggleButton 공통 컴포넌트로 사용하여 ui 폴더로 이동
- 폴더 이동으로 인한 경로 수정
- constant 폴더 확장성 고려하여 constants 폴더로 수정

faber-h committed:
refactor: isPlaying 사용자 인터랙션으로 재생 시작해야 하므로 초기 값 false로 수정

Eun0713 reviewed (approved):
LGTM🍀

devjaeyoon reviewed (approved):
LGTM!
